### PR TITLE
Thumbnails tracked on-disk only

### PR DESCRIPTION
**Pull request by tbarsballe**
Update to thumbnails to use only the filesystem (Based on AppConfiguration.getCacheDir()) to track the thumbnails. All usage of Metadata for tracking thumbnails has been removed.

### CODE REVIEW

File: geoserver/webapp/src/main/java/com/boundlessgeo/geoserver/AppConfiguration.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public void setServletContext(ServletContext servletContext) {
      * Returns the location of the composer cache directory
      * @return Path to the composer cache directory
      */
-    public String getCacheDir() {
+    public String cacheDir() {
```
Comment (tbarsballe) [inline geoserver/webapp/src/main/java/com/boundlessgeo/geoserver/AppConfiguration.java]: To match cacheFile()

File: geoserver/webapp/src/test/java/com/boundlessgeo/geoserver/catalog/ThumbnailInvalidatingCatalogListenerTest.java [hunk 2]
```diff
@@ -1,37 +0,0 @@
...
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
-import static com.boundlessgeo.geoserver.catalog.ThumbnailInvalidatingCatalogListener.updateMetadataMap;
-
-import org.geoserver.catalog.MetadataMap;
-import org.junit.Test;
-
-import com.boundlessgeo.geoserver.api.controllers.Metadata;
-
-public class ThumbnailInvalidatingCatalogListenerTest {
```
Comment (tbarsballe) [inline geoserver/webapp/src/test/java/com/boundlessgeo/geoserver/catalog/ThumbnailInvalidatingCatalogListenerTest.java]: This only tested metadata map methods. Everything involving the filesystem is by necessity in AppIntegrationTest

File: geoserver/webapp/src/main/java/com/boundlessgeo/geoserver/api/controllers/ThumbnailController.java [hunk 3]
```diff
@@ -133,29 +133,16 @@ public ThumbnailController(GeoServer geoServer) {
      * @throws Exception
      */
     public HttpEntity<byte[]> get(WorkspaceInfo ws, PublishedInfo layer, boolean hiRes) throws Exception {
-        String path = Metadata.thumbnail(layer);
+        String path = thumbnailFilename(layer, hiRes);
```
Comment (jodygarnett) [inline geoserver/webapp/src/main/java/com/boundlessgeo/geoserver/api/controllers/ThumbnailController.java]: Just a note that hiRes is not a true/false it is a 1x, 2x, 4x at present. Not worth changing now, expect to change in the next year when the client changes.

File: geoserver/webapp/src/main/java/com/boundlessgeo/geoserver/catalog/ThumbnailInvalidatingCatalogListener.java [hunk 4]
```diff
@@ -26,14 +29,20 @@
...
+    ThumbnailController controller;
     
     /** Flag to prevent recursion during layer, layergroup events */
     private boolean modifying = false;
     
-    public ThumbnailInvalidatingCatalogListener(Catalog catalog) {
-        this.catalog = catalog;
+    public ThumbnailInvalidatingCatalogListener() { }
+    
+    @PostConstruct
```
Comment (jodygarnett) [inline geoserver/webapp/src/main/java/com/boundlessgeo/geoserver/catalog/ThumbnailInvalidatingCatalogListener.java]: That is tricky.

**Comment by jodygarnett** [comment]
Much cleaner implementation, merge when ready.

**Pull request merged**